scripts/exgdas_enkf_earc.sh

```diff
@@ -143,53 +140,154 @@ fi
 
 ###############################################################
 # ENSGRP 0 also does clean-up
-if [ "${ENSGRP}" -eq 0 ]; then
+###############################################################
+if [[ "${ENSGRP}" -eq 0 ]]; then
+    function remove_files() {
+        # TODO: move this to a new location
+        local directory=$1
+        shift
+        if [[ ! -d ${directory} ]]; then
+            echo "No directory ${directory} to remove files from, skiping"
+            return
+        fi
+        local exclude_list=""
+        if (($# > 0)); then
+            exclude_list=$*
+        fi
+        local file_list
+        declare -a file_list
+        # Suppress warnings about chained commands suppressing exit codes
+        # shellcheck disable=SC2312
+        readarray -t file_list < <(find -L "${directory}" -type f)
+        if (( ${#file_list[@]} == 0 )); then return; fi
+        for exclude in ${exclude_list}; do
+            echo "Excluding ${exclude}"
+            declare -a file_list_old=("${file_list[@]}")
+            # Suppress warnings about chained commands suppressing exit codes
+            # shellcheck disable=SC2312
+            readarray file_list < <(printf -- '%s\n' "${file_list_old[@]}" | grep -v "${exclude}")
+            if (( ${#file_list[@]} == 0 )); then return; fi
+        done
+
+        for file in "${file_list[@]}"; do
+            rm -f "${file}"
+        done
+        # Remove directory if empty
+        rmdir "${directory}" || true
+    }
 
     # Start start and end dates to remove
-    GDATEEND=$(${NDATE} -"${RMOLDEND_ENKF:-24}"  "${CDATE}")
-    GDATE=$(${NDATE} -"${RMOLDSTD_ENKF:-120}" "${CDATE}")
+    GDATEEND=$(${NDATE} -"${RMOLDEND_ENKF:-24}"  "${PDY}${cyc}")
+    GDATE=$(${NDATE} -"${RMOLDSTD_ENKF:-120}" "${PDY}${cyc}")
+
     while [ "${GDATE}" -le "${GDATEEND}" ]; do
 
-        gPDY=$(echo "${GDATE}" | cut -c1-8)
-        gcyc=$(echo "${GDATE}" | cut -c9-10)
-
-        # Loop over GDAS and GFS EnKF directories separately.
-        clist="gdas gfs"
-        for ctype in ${clist}; do
-            COMIN_ENS="${ROTDIR}/enkf${ctype}.${gPDY}/${gcyc}"
-            if [ -d "${COMIN_ENS}" ]; then
-                rocotolog="${EXPDIR}/logs/${GDATE}.log"
-                if [ -f "${rocotolog}" ]; then
-                    set +e
-                    testend=$(tail -n 1 "${rocotolog}" | grep "This cycle is complete: Success")
-                    rc=$?
-                    set_strict
-                    if [ "${rc}" -eq 0 ]; then
-                        # Retain f006.ens files.  Remove everything else
-                        for file in $(ls "${COMIN_ENS}" | grep -v f006.ens); do
-                            rm -rf "${COMIN_ENS}"/"${file}"
+        gPDY="${GDATE:0:8}"
+        gcyc="${GDATE:8:2}"
+
+        if [[ -d ${COM_TOP} ]]; then
+            rocotolog="${EXPDIR}/logs/${GDATE}.log"
```

Comment:
I strongly dislike that this script is sourcing from the rocoto logs but this can be addressed in a different issue. Just wanted to state that. :)

Reply:
So am I. I think we already have an issue open to fix it.